lib/api/apiUtils/object/versioning.js

```diff
@@ -401,6 +403,18 @@ function overwritingVersioning(objMD, metadataStoreParams) {
     metadataStoreParams.creationTime = objMD['creation-time'];
     metadataStoreParams.lastModifiedDate = objMD['last-modified'];
     metadataStoreParams.updateMicroVersionId = true;
+
+    // update restore
+    const days = objMD.archive?.restoreRequestedDays;
+    const now = Date.now();
+    metadataStoreParams.archive = {
+        archiveInfo: objMD.archive?.archiveInfo,
+        restoreRequestedAt: objMD.archive?.restoreRequestedAt,
+        restoreRequestedDays: objMD.archive?.restoreRequestedDays,
```

Comment:
LGTM, just a small question: can these values not exist in objectMD when put object version? do we need to throw an error if objectMD.archive doesn't exist?

Reply:
We check earlier that these value does not exist: https://github.com/scality/cloudserver/pull/4539/files#diff-f2ca335d4dd243e368c6c2752d4aa2ac20fe772d94b2a21cd877a235a7ef0850R161 
Note: we do not use `putObjectVersion` to update the restoration period.

We already return an error if `objectMD.archive` doesn't exist https://github.com/scality/cloudserver/pull/4539/files#diff-f2ca335d4dd243e368c6c2752d4aa2ac20fe772d94b2a21cd877a235a7ef0850R160

Reply:
ok I see. right,  [this check](https://github.com/scality/cloudserver/pull/4539/files#diff-f2ca335d4dd243e368c6c2752d4aa2ac20fe772d94b2a21cd877a235a7ef0850R161) happens before here